junit-vintage-engine/src/main/java/org/junit/vintage/engine/VintageTestEngine.java

```diff
@@ -70,17 +86,107 @@ public void execute(ExecutionRequest request) {
 		engineExecutionListener.executionStarted(engineDescriptor);
 		RunnerExecutor runnerExecutor = new RunnerExecutor(engineExecutionListener,
 			engineDescriptor.getTestSourceProvider());
-		executeAllChildren(runnerExecutor, engineDescriptor);
+
+		final ConfigurationParameters configurationParameters = request.getConfigurationParameters();
+		final boolean parallelExecutionEnabled = configurationParameters.getBoolean(PARALLEL_ENABLED_CONFIG).orElse(
+			false);
+
+		try (CloseableExecutor executor = parallelExecutionEnabled ? new ParallelExecutor(configurationParameters)
+				: new SerialExecutor()) {
```

Comment:
Please extract the `CloseableExecutor` creation into a method.

Reply:
This is done.